docs/custom_promise_types/gpg_keys_using_lib.py

```diff
@@ -0,0 +1,154 @@
+"""Custom Promise Type module for adding missing gpg keys
+
+Given some GPG user id values (see man page for all possiblities, there are many)
+and the ascii value of a key, ensure that the key is present in the given GPG homedir.
+Typically the user id will be a key fingerprint or an email address.
+
+Installation:
+  cp gpg_keys_using_lib.py $(sys.workdir)/modules/promises
```

Comment:
OK, but if you are using masterfiles, the preferred location is `masterfiles/modules/promises/`, which will be copied there automatically.